### ISSUE TITLE
attention mask

### ISSUE THREAD

**Issue by zhhao1** [HuangLK/llama-deepspeed#27]
在feeder.py里给模型提供的是因果mask，但是没有提供pad mask，这个地方似乎需要改进一下。

**Comment by HuangLK** [comment]
pad不计算loss，应该不影响

**Comment by zhhao1** [comment]
> pad不计算loss，应该不影响
但是不对pad掩玛的话，计算attention的时候，pad也会参与计算，由于没有mask为-inf，softmax输出不接近0，其对应的权重也会影响注意力层的输出吧

**Comment by HuangLK** [comment]
哦 在这里pad是拼在后面的，mask只让看前面的token

**Comment by zhhao1** [comment]
恩恩，现在只有因果mask，看不到前面的token，但是有一部分是pad，应该也是不能看到，不参与中间计算的，但是输入少了对pad这部分的掩玛，这个地方你看是这个样子不。

**Comment by zhhao1** [comment]
打扰了。。。。
